app/src/main/java/emu/skyline/AppDialog.kt

```diff
@@ -101,6 +154,66 @@ class AppDialog : BottomSheetDialogFragment() {
             shortcutManager.requestPinShortcut(info.build(), null)
         }
 
+        val saveFolderPath = savesFolderRoot + item.titleId
+        val saveExists = File(saveFolderPath).exists()
+
+        binding.deleteSave.isEnabled = saveExists
+        binding.deleteSave.setOnClickListener {
+            AlertDialog.Builder(requireContext())
+                .setTitle(getString(R.string.delete_save_confirmation_message))
+                .setMessage(getString(R.string.action_irreversible))
+                .setPositiveButton(getString(R.string.yes)) { dialogInterface, _ ->
+                    File(saveFolderPath).deleteRecursively()
+                    binding.deleteSave.isEnabled = false
+                    binding.exportSave.isEnabled = false
+                    dialogInterface.dismiss()
+                }.setNegativeButton(getString(R.string.no)) { dialogInterface, _ ->
+                    dialogInterface.cancel()
+                }.show()
+        }
+
+        binding.importSave.setOnClickListener {
+            documentPicker.launch(arrayOf("application/zip"))
+        }
+
+        binding.exportSave.isEnabled = saveExists
+        binding.exportSave.setOnClickListener {
+            CoroutineScope(Dispatchers.IO).launch {
```

Comment:
CoroutineScope(Dispatchers.IO).launch(Dispatchers.IO) resolves warnings about potential thread starvation

Reply:
No need to do this, I don't even know why Android Studio would even warn about this here.

Reply:
It doesn't warn about it here. It warns that `createNewFile` and `FileOutputStream` are `possibly blocking calls in non-blocking context that could lead to thread starvation`. Android Studio actually wants you to wrap both in withContext to suspend the thread until the method is complete, so this is the "suppression" version.